### PR TITLE
[RUM-16753] Add TTL-based rate limiting to RemoteConfigurationProvider foreground refresh

### PR DESCRIPTION
**Pull request by saraSr5**
### What and why?

Adds TTL-based rate limiting to the foreground refresh introduced in RUM-16711. Without a TTL, the RemoteConfigurationProvider would make a CDN request on every app foreground transition, even when the app was only briefly backgrounded. This reduces unnecessary network requests by suppressing foreground syncs that happen within 5 minutes of the last successful revalidation, following the RFC's recommended caching strategy.

### How?

- dateProvider: DateProvider is injected in RemoteConfigurationProvider.init following the same pattern as DatadogCore, allowing tests to control time without waiting.
- lastSyncDate: Date? (protected by @ReadWriteLock) tracks when the last successful sync occurred.
- The foreground observer checks dateProvider.now.timeIntervalSince(lastSyncDate) < minimumSyncInterval before firing a sync — if within 5 minutes, it skips.
- lastSyncDate is updated after a successful 200 response (disk write) and after a 304 (cache still valid). Failed syncs do not update it, so the next foreground attempt is not blocked.
- minimumSyncInterval = 300 seconds (5 minutes) is a private constant per the RFC.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes
- [ ] Add Objective-C interface for public APIs - see our [guidelines](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/3157787243/RFC+-+Modular+Objective-C+Interface#Recommended-solution) (internal) 
- [ ] Run `make api-surface` when adding new APIs

### CODE REVIEW

**Comment by saraSr5** [comment]
@codex review

File: DatadogCore/Sources/Core/RemoteConfigurationProvider.swift [hunk 1]
```diff
@@ -116,6 +129,7 @@ internal final class RemoteConfigurationProvider {
             case .success(let (http, data)):
                 // 1. Not Modified — existing persisted configuration is still valid.
                 if http.statusCode == 304 {
+                    self.lastSyncDate = self.dateProvider.now
```
[the remark this answers is not in the document]
Comment (saraSr5) [reply DatadogCore/Sources/Core/RemoteConfigurationProvider.swift:134]: This scenario, about 304 with no cached JSON, requires the .etag file to exist without a corresponding .json file. Since both are always written together atomically on a successful 200 response, and never deleted independently, this state cannot occur in normal SDK operation. 
I wouldn't add defensive code for a not so possible condition

[the remark this answers is not in the document]
Comment (saraSr5) [reply DatadogCore/Sources/Core/RemoteConfigurationProvider.swift:134]: The dateProvider abstraction already handles this. In tests it's use RelativeDateProvider which only advances forward. 
SystemDateProvider uses Date() which follows the system clock. A large NTP rollback is an extreme edge case that would affect many other SDK components beyond the TTL. 
I wouldn't add special handling for this scenario.

**Review by arroz — APPROVED** [review]
Looks good! 👍
